script.js

```diff
@@ -0,0 +1,265 @@
+
+moviesRating(array);
+
+//Simple approach
+const moviesRatingSimple = (array) => {
+  return array.filter((movie) => movie.rating > 8.5);
+};
+
+moviesRatingSimple(array);
+
+
+
+// Sort the movies by rating in descending order.
+
+const moviesDescending = (array) => {
+  return array.map((movie) => movie.rating).sort((a, b) => (a > b ? -1 : 1));
+};
+
+moviesDescending(array);
+
+
+
+// Create an array of movies released after the year 2000.
+const getMoviesReleasedAfter2000 = (array) => {
+  return array.filter((movie) => movie.year > 2000).map((movie) => movie.title);
+};
+
+getMoviesReleasedAfter2000(array);
+
+
+
+// Sort the movies by year of release in ascending order.
+const moviesAscending = (array) => {
+  return array.map((movie) => movie.year).sort((a, b) => (a > b ? 1 : -1));
+};
+
+moviesAscending(array);
+
+
+
+// Create an object that counts the number of movies for each genre.
+const countMoviesByGenre = (array) => {
+  return array.reduce((acc, movie) => {
+    movie.genre.forEach(genre => {
+      acc[genre] = (acc[genre] || 0) + 1;
+    });
+    return acc;
+  }, {});
+};
+
+countMoviesByGenre(array);
+
+
+
+// Check if all movies have a rating higher than 7.
+const moviesRatingHigher7 = (array) => {
+  return array.every((movie) => movie.rating > 7);
+};
+
+moviesRatingHigher7(array);
+
+
+
+// Calculate the average rating of all movies.
+const averageRating = (array) => {
+  return array.reduce((acc, movie, __, { length }) => acc + movie.rating / length, 0);
+};
+
+averageRating(array);
+
+
+
+// Create an array of objects with only the title and rating of each movie.
+const arrayTitlesRating = (array) => {
+  return array.map(({ title, rating }) => ({ title, rating }));
+};
+
+arrayTitlesRating(array);
+
+
+
+// Check if there is at least one movie in the "Sci-Fi" genre.
+const movieSciFigenre = (array) => {
+  return array.some((movie) => movie.genre.includes("Sci-Fi"));
+};
+
+movieSciFigenre(array);
+
+
+
+// Check if there is at least one movie directed by "Quentin Tarantino".
+const directedByTarantino = (array) => {
+  return array.some((movie) => movie.director === "Quentin Tarantino");
+};
+
+directedByTarantino(array);
+
+
+
+// Check if all movies were released after 1990.
+const realeseAfter1990 = (array) => {
+  return array.every((movie) => movie.year > 1990);
+};
+
+realeseAfter1990(array);
+
+
+
+// Find the titles of all movies directed by "Christopher Nolan" that have a rating higher than 8.5, sorted by rating in descending order.
+const titlesDirectedByNolan = (array) => {
+  return array
+    .filter(
+      (movie) => movie.director === "Christopher Nolan" && movie.rating > 8.5,
+    )
+    .sort((a, b) => (a.rating > b.rating ? -1 : 1))
+    .map((movie) => movie.title);
+};
+
+titlesDirectedByNolan(array);
+
+
+
+// Calculate the average rating of movies released before the year 2000 that belong to the "Drama" genre.
+const averageRatingMovies = (array) => {
+  const filteredMovies = array.filter(
```

Comment:
You have to do this without defining local variables for it.
You can chain them without the variables.